packages/-ember-data/tests/unit/store/adapter-interop-test.js

```diff
@@ -116,6 +116,64 @@ module('unit/store/adapter-interop - Store working with a Adapter', function (ho
     });
   });
 
+  test('Coalesced Store#findRecord requests retain the `include` adapter option in the snapshots passed to adapter#findMany and adapter#findRecord', function (assert) {
+    const includedResourcesForIds = {
+      1: 'someResource',
+      2: 'differentResource',
+      3: 'anotherResource',
+    };
+
+    assert.expect(Object.keys(includedResourcesForIds).length);
+
+    const ApplicationAdapter = Adapter.extend({
+      groupRecordsForFindMany(store, snapshots) {
+        // this ensures that `findRecord` is also called
+        return [snapshots.slice(0, -1), snapshots.slice(-1)];
+      },
+
+      findMany(store, type, ids, snapshots) {
+        snapshots.forEach((snapshot) => {
+          assert.equal(
+            snapshot.include,
+            includedResourcesForIds[snapshot.id],
+            `Snapshot #${snapshot.id} retains the 'include' adapter option in #findMany`
+          );
+        });
+        return resolve({
+          data: snapshots.map(({ id }) => ({ id, type: type.modelName })),
+        });
+      },
+
+      findRecord(store, type, id, snapshot) {
+        assert.equal(
+          snapshot.include,
+          includedResourcesForIds[snapshot.id],
+          `Snapshot #${snapshot.id} retains the 'include' adapter option in #findRecord`
+        );
+
+        return resolve({
+          data: { id, type: type.modelName },
+        });
+      },
+
+      coalesceFindRequests: true,
+    });
+
+    this.owner.register('adapter:application', ApplicationAdapter);
+    this.owner.register('serializer:application', JSONAPISerializer.extend());
+    this.owner.register('model:test', Model.extend());
+
+    let store = this.owner.lookup('service:store');
+
+    return run(() => {
```

Comment:
we should make the test async and await the all, no need for the `run` wrapper at all anymore.

Reply:
Oh, yes, of course

Reply:
🤔  I migrate to async/await only for this test ? Seems like the whole file still rely on run() :(

Reply:
Just the new test for now, but test cleanup PRs are always welcome!